core/src/main/java/org/testcontainers/containers/JibDockerClient.java

```diff
@@ -0,0 +1,82 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectImageResponse;
+import com.github.dockerjava.api.command.LoadImageCallback;
+import com.google.cloud.tools.jib.api.DockerClient;
+import com.google.cloud.tools.jib.api.ImageDetails;
+import com.google.cloud.tools.jib.api.ImageReference;
+import com.google.cloud.tools.jib.http.NotifyingOutputStream;
+import com.google.cloud.tools.jib.image.ImageTarball;
+import com.google.common.io.ByteStreams;
+import lombok.Cleanup;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.images.RemoteDockerImage;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class JibDockerClient implements DockerClient {
```

Comment:
`@Unstable`, to allow us to merge it, but have more flexibility to improve it in the future?

Reply:
done!